src/Symfony/Component/Config/Resource/GlobResource.php

```diff
@@ -167,28 +167,31 @@ function (\SplFileInfo $file, $path) {
             throw new \LogicException(sprintf('Extended glob pattern "%s" cannot be used as the Finder component is not installed.', $this->pattern));
         }
 
-        $finder = new Finder();
         $regex = Glob::toRegex($this->pattern);
         if ($this->recursive) {
             $regex = substr_replace($regex, '(/|$)', -2, 1);
         }
 
         $prefixLen = \strlen($this->prefix);
-        foreach ($finder->followLinks()->sortByName()->in($this->prefix) as $path => $info) {
-            $normalizedPath = str_replace('\\', '/', $path);
-            if (!preg_match($regex, substr($normalizedPath, $prefixLen)) || !$info->isFile()) {
-                continue;
-            }
-            if ($this->excludedPrefixes) {
-                do {
-                    if (isset($this->excludedPrefixes[$dirPath = $normalizedPath])) {
-                        continue 2;
-                    }
-                } while ($prefix !== $dirPath && $dirPath !== $normalizedPath = \dirname($dirPath));
-            }
 
-            yield $path => $info;
-        }
+        yield from (new Finder())
+            ->followLinks()
+            ->filter(function (\SplFileInfo $info) use ($regex, $prefixLen, $prefix) {
+                $normalizedPath = str_replace('\\', '/', $info->getPathname());
+                if (!$info->isFile() || !preg_match($regex, substr($normalizedPath, $prefixLen))) {
```

Comment:
Did you swap the operands on purpose? isn't preg_match faster than isFile?

Reply:
Good catch! I made some tests and It seems that `preg_match` + `substr` is faster than `isFile`.